compiler/block.py

```diff
@@ -440,16 +440,29 @@ def __init__(self, node):
     BlockVisitor.__init__(self)
     self.is_generator = False
     node_args = node.args
-    args = [a.arg for a in node_args.args]
+    args = []
+    for arg in node_args.args:
+      if isinstance(arg, ast.Tuple):
+        args.append([a.arg for a in arg.elts])
+      else:
+        args.append(arg.arg)
     if node_args.vararg:
       args.append(node_args.vararg.arg)
     if node_args.kwarg:
       args.append(node_args.kwarg.arg)
     for i, name in enumerate(args):
-      if name in self.vars:
-        msg = "duplicate argument '{}' in function definition".format(name)
-        raise util.ParseError(node, msg)
-      self.vars[name] = Var(name, Var.TYPE_PARAM, arg_index=i)
+      if isinstance(name, list):
+        self.add_vars(i, 'τ'.join(name), node)
```

Comment:
@trotterdylan - Since we are passing the Tuple as a single argument, we'll need some sort of Vars to hold the values inside the function as we need to **Tie** these arguments to the respective variables/arguments. Instead of pseudo Vars, what do you suggest?

Another approach could be to unpack them while constructing the arguments to the function itself and pass all the unpacked values as directly. 

Let me know!